### PR TITLE
Improve code coverage, a little cleanup, and some more overloads.

### PR DESCRIPTION
**Pull request by zippy1981**
## Type of Change

 - [X] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [X] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [X] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [X] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Improve code coverage of Size.cs and refactored it a little bit based on that.

### Approach
I banged on the keyboard until a PR happened.

### Commands to test
You know with all the time I spend writing unit tests for Size.cs you think I'd actually use a cmdlet that returned Size properties

### CODE REVIEW

File: bin/projects/dbatools/dbatools/Utility/Size.cs [hunk 1]
```diff
@@ -179,34 +173,23 @@ public Size(long Byte)
...
+            if (obj is Size)
             {
-                if (Byte == ((Size)obj).Byte) { return 0; }
-                if (Byte < ((Size)obj).Byte) { return -1; }
-
-                return 1;
+                return CompareTo((Size) obj);
             }
-            catch { return 0; }
+            throw new ArgumentException(String.Format("Cannot compare a {0} to a {1}", typeof(Size).FullName, obj.GetType().FullName));
```
Comment (zippy1981) [inline bin/projects/dbatools/dbatools/Utility/Size.cs]: I think its perfectly ok for this to throw an exception. If you disagree, create a test that shows this exception happening in any use case that's reasonable.

**Comment by potatoqualitee** [comment]
Will leave this up to @FriedrichWeinmann to merge, thanks zippy!

**Comment by potatoqualitee** [comment]
:bananadance: thank you both!

**Comment by zippy1981** [comment]
@FriedrichWeinmann Your're welcome.

**Pull request merged**